src/compiler/checker.ts

```diff
@@ -8337,10 +8340,23 @@ module ts {
         }
 
         function checkTypeReferenceNode(node: TypeReferenceNode) {
+            // Check if the type reference is using strict mode keyword
+            // Example:
+            //      class C {
+            //          foo(x: public){}  // Error.
+            //      }
+            if (node.typeName.kind === SyntaxKind.Identifier && (<Identifier>node.typeName).strictModeKind) {
+                let typeName = <Identifier>node.typeName;
+                let nameText = declarationNameToString(typeName);
+                reportStrictModeGrammarErrorInClassDeclaration(typeName, Diagnostics.Type_expected_0_is_a_reserved_word_Class_definitions_are_automatically_in_strict_mode, nameText) ||

```

Comment:
This needs to be "Type expected. '{0}' is a reserved word *in strict mode*. Class blah blah blah"